pkg/scheduler/backend/queue/scheduling_queue.go

```diff
@@ -599,7 +605,11 @@ func (p *PriorityQueue) Activate(logger klog.Logger, pods map[string]*v1.Pod) {
 	for _, pod := range pods {
 		if p.activate(logger, pod) {
 			activated = true
+			continue
 		}
+
+		// If this pod is in-flight, register the activation event so that the pod will be requeued when it comes back.
+		p.activeQ.addEventsIfPodInFlight(nil, pod, []framework.ClusterEvent{framework.EventForceActivate})
```

Comment:
Wouldn't this event match every pod that is in-flight? Or is that desired?

Also, for future reference, can you summarize why this change is needed?

Reply:
Oops, good catch, I forgot putting another change that must be needed on handling this activation event.

Reply:
> can you summarize why this change is needed?

Please add this to the PR description.

Reply:
Done